crates/bitwarden-vault/src/tree.rs

```diff
@@ -0,0 +1,275 @@
+use std::collections::HashMap;
+use std::fmt::Debug;
+use uuid::Uuid;
+
+pub trait TreeItem: Clone + Debug {
+    fn id(&self) -> Option<Uuid>;
```

Comment:
suggestion: since you are using `unwrap` on this that must mean that you **_really_** don't expect this to be `None`, in which case I'd suggest removing the option completely
```suggestion
    fn id(&self) -> Uuid;
```